models/repo_activity.go

```diff
@@ -91,12 +92,20 @@ func GetActivityStatsTopAuthors(repo *Repository, timeFrom time.Time, count int)
 		return nil, nil
 	}
 	users := make(map[int64]*ActivityAuthorData)
-	for k, v := range code.Authors {
-		if len(k) == 0 {
+	var unknownUserID int64
+	unknownUserAvatarLink := NewGhostUser().AvatarLink()
+	for _, v := range code.Authors {
+		if len(v.Email) == 0 {
 			continue
 		}
-		u, err := GetUserByEmail(k)
+		u, err := GetUserByEmail(v.Email)
 		if u == nil || IsErrUserNotExist(err) {
+			unknownUserID = unknownUserID - 1
```

Comment:
This will create multiple entries for rows that perhaps belong to the same user. Is that OK?

Reply:
They are already grouped by unique email addresses, this resolves to Gitea users and groups by user id, this is just negative unique id given for each unresolved email

Reply:
I think we can group all ghost users as one user, but not every ghost has only 1 count.

Reply:
It's not really ghost user, it's git author, I just display ghost users icon for it

Reply:
But for the git user who has the same email, shouldn't we think it's the same user?

Reply:
it already does, authors with different email addresses will be treated as different users